### ISSUE TITLE
CryptoMatte AOV support

### ISSUE THREAD

**Issue by thiagodesul** [LuxCoreRender/LuxCore#285]
Hello! Are you planning to add support to CryptoMatte in Lux?

If yes, is it possible to save Cryptomatte files in a separated .EXR file as Redshift usually does. It's better for our comp artists to have beauty and data passes in a different file.  

Thank you!

**Comment by Dade916** [comment]
There has been talking about CryptoMatte in LuxCore but I'm not sure when it is going to happen, we have a very long TODO list at the moment.

**Comment by juangea** [comment]
> There has been talking about CryptoMatte in LuxCore but I'm not sure when it is going to happen, we have a very long TODO list at the moment.

Yes, but allow me to say that you are doing great, a super wonderful job, and we will try to attract more users, and also donors and collaborators :)

I know this is off-topic, you can delete this comment, but I wanted to tell this for your answer :)

**Comment by thiagodesul** [comment]
Agreed!!

**Comment by AlexeyAdamitsky** [comment]
I posted some bounty on the task a month ago. I don't know how it works but I assumed it would automatically reflect in the thread title. If not it would be nice if it's get added as for the other topics. Maybe it'll pique someone's interest to work on the feature.

**Comment by Dade916** [comment]
It should now.

**Comment by RumbleballTheReal** [comment]
I was looking into this today. I have not much knowledge of rendering yet. Would it be neccessary to do a deep integration into the rendering pipeline or would it be enough to tell LuxCore to just return mattes for a defined set of objects and then create the Cryptomatte from all single renders.

Blender::Cycles has a deep integration of Cryptomatte in its kernel.

**Comment by juangea** [comment]
Cryptomatte are not mattes itself, it's a bunch of metadata that allows the cryptomatte node (or interpreter) to extract whatever matte we need in compositing, without pre-setting anything in the scene, you just enable cryptomatte and you are good to go, then you can extract fine mattes in Blender compositor, or Fusion or Nuke or whatever other software with cryptomatte support :)

**Comment by RumbleballTheReal** [comment]
@juangea its not about reading the data out, its about retriving and writing the data with LuxCore

**Comment by juangea** [comment]
I know 👍 
I was answering to this: 

`would it be enough to tell LuxCore to just return mattes for a defined set of objects and then create the Cryptomatte from all single renders.`

But I may have misinterpreted what you meant there :)

**Comment by juangea** [comment]
This may interest you, it' the Eevee implementation for Cryptomatte :)

https://developer.blender.org/D9165

**Comment by mischterlampe** [comment]
Are there any advantages for luxcore renderings, if you can create cyptomattes with Eevee? 
I think an integration of cryptomatte into luxcore is essential for post production and professional work. Someone who has worked with cryptomattes won't want to work without them. I think the options cycles has are good. You can pick a matte for every single geometry, for every material and for the whole object. 
Does anyone has a clue how long an integration would roughly take?
